cpp/src/io/parquet/parquet_common.hpp

```diff
@@ -86,11 +86,12 @@ enum class Encoding : uint8_t {
   GROUP_VAR_INT           = 1,  // Deprecated, never used
   PLAIN_DICTIONARY        = 2,
   RLE                     = 3,
-  BIT_PACKED              = 4,
+  BIT_PACKED              = 4,  // Deprecated
```

Comment:
Question: Could you please clarify how this change relates to the change in the reader?

Reply:
I'm just updating the enum to make it match the current parquet spec, which lists BIT_PACKED as a deprecated encoding.  Can remove if you'd prefer.

Reply:
Can you please add more comments about why/when/how etc. this enum is deprecated? As well as tracking issue (if any)?

Reply:
@ttnghia I did some digging through git blame on the parquet-format README.md.  I found the commit but couldn't find an associated JIRA issue. I added a little more context to the comment.